src/main/java/org/gridsuite/study/server/service/NonEvacuatedEnergyService.java

```diff
@@ -0,0 +1,355 @@
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.HttpStatusCodeException;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.util.UriComponentsBuilder;
+
+import java.io.UncheckedIOException;
+import java.net.URI;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import static org.gridsuite.study.server.StudyConstants.DELIMITER;
+import static org.gridsuite.study.server.StudyConstants.HEADER_USER_ID;
+import static org.gridsuite.study.server.StudyConstants.QUERY_PARAM_RECEIVER;
+import static org.gridsuite.study.server.StudyConstants.QUERY_PARAM_VARIANT_ID;
+import static org.gridsuite.study.server.StudyConstants.SENSITIVITY_ANALYSIS_API_VERSION;
+import static org.gridsuite.study.server.StudyException.Type.DELETE_COMPUTATION_RESULTS_FAILED;
+import static org.gridsuite.study.server.StudyException.Type.NON_EVACUATED_ENERGY_ERROR;
+import static org.gridsuite.study.server.StudyException.Type.NON_EVACUATED_ENERGY_NOT_FOUND;
+import static org.gridsuite.study.server.StudyException.Type.NON_EVACUATED_ENERGY_RUNNING;
+import static org.gridsuite.study.server.utils.StudyUtils.handleHttpError;
+
+/**
+ * @author Franck Lecuyer <franck.lecuyer at rte-france.com>
+ */
+@Service
+public class NonEvacuatedEnergyService {
+
+    static final String RESULT_UUID = "resultUuid";
+
+    private String sensitivityAnalysisServerBaseUri;
+
+    @Autowired
+    private RestTemplate restTemplate;
+
+    private final ObjectMapper objectMapper;
+
+    private final NetworkModificationTreeService networkModificationTreeService;
+
+    @Autowired
+    NonEvacuatedEnergyService(RemoteServicesProperties remoteServicesProperties,
+                              NetworkModificationTreeService networkModificationTreeService,
+                              ObjectMapper objectMapper) {
+        this.sensitivityAnalysisServerBaseUri = remoteServicesProperties.getServiceUri("sensitivity-analysis-server");
+        this.networkModificationTreeService = networkModificationTreeService;
+        this.objectMapper = objectMapper;
+    }
+
+    public void setSensitivityAnalysisServerBaseUri(String sensitivityAnalysisServerBaseUri) {
+        this.sensitivityAnalysisServerBaseUri = sensitivityAnalysisServerBaseUri + DELIMITER;
+    }
+
+    public void assertNonEvacuatedEnergyNotRunning(UUID nodeUuid) {
+        String nonEvacuatedEnergyStatus = getNonEvacuatedEnergyStatus(nodeUuid);
+        if (SensitivityAnalysisStatus.RUNNING.name().equals(nonEvacuatedEnergyStatus)) {
```

Comment:
use a specific NonEvacuatedEnergyStatus for consistency ?

Reply:
Done